components/documents/add-document-modal.tsx

```diff
@@ -175,6 +178,28 @@ export function AddDocumentModal({
     return pdf.numPages;
   }
 
+  const createNotionFileName = () => {
+    // Extract Notion file name from the URL
+    const urlSegments = (notionLink as string).split("/")[3];
+
+    // Remove the last hyphen along with the Notion ID
+    const extractName = urlSegments.replace(/-([^/-]+)$/, "");
+    const notionFileName = extractName.replaceAll("-", " ") || "Notion Link";
+
+    // Check if a Notion file with the same name already exists
+    const existingNotionFiles = documents
```

Comment:
I don't think it's necessary to fetch all existing documents. This could become an expensive operation if the user has a lot of docs. 

It shouldn't matter if the name is the same, the user can always rename the name.

Reply:
I've removed the counter to avoid the expensive operation. :)